armi/bookkeeping/db/database3.py

```diff
@@ -912,7 +1077,16 @@ def _writeParams(self, h5group, comps):
                     attrs[_SERIALIZER_NAME] = paramDef.serializer.__name__
                     attrs[_SERIALIZER_VERSION] = paramDef.serializer.version
                 else:
-                    data = numpy.array(temp)
+                    # check if temp is a jagged array
+                    if any(isinstance(x, (numpy.ndarray, list)) for x in temp):
+                        jagged = len(set([_getShape(x) for x in temp])) != 1
+                    else:
+                        jagged = False
+                    data = (
+                        JaggedArray(temp, paramDef.name)
+                        if jagged
+                        else numpy.array(temp)
+                    )
                     del temp
 
             # Convert Unicode to byte-string
```

Comment:
The next 3 `if` statements should all be mutually exclusive. Might want to make it an `if`/`elif` block.